homeassistant/components/dsmr/sensor.py

```diff
@@ -377,38 +385,38 @@ class DSMRSensorEntityDescription(SensorEntityDescription):
     ),
 )
 
-
-def create_mbus_entity(
-    mbus: int, mtype: int, telegram: Telegram
-) -> DSMRSensorEntityDescription | None:
-    """Create a new MBUS Entity."""
-    if mtype == 3 and hasattr(telegram, f"BELGIUM_MBUS{mbus}_METER_READING2"):
-        return DSMRSensorEntityDescription(
-            key=f"mbus{mbus}_gas_reading",
+SENSORS_MBUS_DEVICE_TYPE: dict[int, tuple[DSMRSensorEntityDescription, ...]] = {
+    MbusDeviceType.GAS: (
+        DSMRSensorEntityDescription(
+            key="gas_reading",
             translation_key="gas_meter_reading",
-            obis_reference=f"BELGIUM_MBUS{mbus}_METER_READING2",
+            obis_reference="MBUS_METER_READING",
+            dsmr_versions={"5B"},
             is_gas=True,
             device_class=SensorDeviceClass.GAS,
             state_class=SensorStateClass.TOTAL_INCREASING,
-        )
-    if mtype == 7 and (hasattr(telegram, f"BELGIUM_MBUS{mbus}_METER_READING1")):
-        return DSMRSensorEntityDescription(
-            key=f"mbus{mbus}_water_reading",
+        ),
+    ),
+    MbusDeviceType.WATER: (
+        DSMRSensorEntityDescription(
+            key="water_reading",
             translation_key="water_meter_reading",
-            obis_reference=f"BELGIUM_MBUS{mbus}_METER_READING1",
+            obis_reference="MBUS_METER_READING",
+            dsmr_versions={"5B"},
```

Comment:
Same, also allow (at least) 5

Reply:
Agree. There is no point in limiting this here by version. Removed.